src/Model/Event.php

```diff
@@ -577,75 +651,84 @@ public static function prepareListForTemplate(array $event_result): array
 	{
 		$event_list = [];
 
-		$last_date = '';
-		$fmt = DI::l10n()->t('l, F j');
 		foreach ($event_result as $event) {
-			$item = Post::selectFirst(['plink', 'author-name', 'author-network', 'author-id', 'author-avatar', 'author-link', 'private', 'uri-id'], ['id' => $event['itemid']]);
-			if (!DBA::isResult($item)) {
-				// Using default values when no item had been found
-				$item = ['plink' => '', 'author-name' => '', 'author-avatar' => '', 'author-link' => '', 'private' => Item::PUBLIC, 'uri-id' => ($event['uri-id'] ?? 0)];
-			}
+			$event_list[] = static::prepareForItem($event);
+		}
 
-			$event = array_merge($event, $item);
+		return $event_list;
+	}
 
-			$start = DateTimeFormat::local($event['start'], 'c');
-			$j     = DateTimeFormat::local($event['start'], 'j');
-			$day   = DateTimeFormat::local($event['start'], $fmt);
-			$day   = DI::l10n()->getDay($day);
+	/**
+	 * Convert an one event in an array which could be used by the events template.
+	 *
+	 * @param array $event Event query array.
+	 * @return array Event array for the template.
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws \ImagickException
+	 */
+	public static function prepareForItem(array $event): array
+	{
+		$fmt = DI::l10n()->t('l, F j');
 
-			if ($event['nofinish']) {
-				$end = null;
-			} else {
-				$end = DateTimeFormat::local($event['finish'], 'c');
-			}
+		$item = Post::selectFirst(['plink', 'author-name', 'author-network', 'author-id', 'author-avatar', 'author-link', 'private', 'uri-id'], ['id' => $event['itemid']]);
+		if (!DBA::isResult($item)) {
```

Comment:
`Post::selectFirst` doesn't return a statement so the `isResult` call is unwarranted here.
```suggestion
		$item = Post::selectFirst(['plink', 'author-name', 'author-network', 'author-id', 'author-avatar', 'author-link', 'private', 'uri-id'], ['id' => $event['itemid']]);
		if (!$item) {
```